pylint/interfaces.py

```diff
@@ -46,7 +59,9 @@ def is_implemented_by(cls, instance):
         return implements(instance, cls)
 
 
-def implements(obj: "Interface", interface: Tuple[type, type]) -> bool:
+def implements(
+    obj: "Interface", interface: Union[Type["Interface"], Tuple[Type["Interface"], ...]]
```

Comment:
```suggestion
    obj: BaseChecker, interface: Union[Type["Interface"], Tuple[Type["Interface"], ...]]
```

@Pierre-Sassoulas Since the previous comment was merged I can't suggest on it.

I think it should be BaseChecker actually. We use this to see if checkers implement interfaces. However, since the checkers don't actually subclass `Interface` this will start throwing errors when we add typing to checkers that call `implements`.

The use of `__implements__` seems to be quite outdated anyway, see this rejected [pep](https://www.python.org/dev/peps/pep-0245/)

Reply:
I added that typing change because it felt like a quick typing fix, but it seems it's becoming a bigger refactor now. It's going to be a problem later on though, so I guess we could merge as is.

When the pip has functions in CamelCase you know it's old 😄. Would you suggest to change the implementation of interfaces later ? Maybe we could we use duck typing and ``has_attr`` or similar instead ? Do you want me to open an issue ?

Reply:
Yeah I think with my latest suggestion we have fixed the issue here! PRs like this (small change + small typing update) are actually really useful in our typing efforts.

I think opening an issue doesn't make a lot of sense right now: it will just clutter the issue list and we probably won't have any actual useful discussion going on. Let's add this to our mental todo list 😄